test/curve_analysis/test_baseclass.py

```diff
@@ -226,12 +231,43 @@ def test_end_to_end_single_function(self):
         result = analysis.run(test_data)
         self.assertExperimentDone(result)
 
+        curve_data = result.artifacts("curve_data").data
```

Comment:
Alternatively you can create reference dataframe and directly compare with
```python
ref_data = ScatterTable.from_dataframe(ref_frame)
self.assertEqual(curve_data, ref_data)
```
I guess this is more strict.

Reply:
That's true, but the comparison can't be made directly because the `ScatterTable` contains random `yval` and `yerr` values that would have to be excluded from the comparison (unless we save the full `ScatterTable` for a particular seed in the test code), so I thought the two different approaches would be about the same.

Reply:
Here is the code. You can run data_processor locally. 

```python
        curve_data = result.artifacts("curve_data").data

        processed_y = data_processor(test_data.data())
        ref_raw_subset = pd.DataFrame.from_dict(
            {
                "xval": x,
                "yval": unp.nominal_values(processed_y),
                "yerr": unp.std_devs(processed_y),
                "series_name": np.full(100, "test"),
                "series_id": np.full(100, 0),
                "category": np.full(100, "raw"),
                "shots": np.full(100, 10000),
                "analysis": np.full(100, "CurveAnalysis"),
            }
        )
        self.assertEqual(
            ScatterTable.from_dataframe(ref_raw_subset),
            curve_data.filter(category="raw"),
        )
```

But I found that it's difficult to test `formatted` and `fitted` subset. Currently test is performed against everything we can predict, but this is likely just testing the behavior of ScatterTable itself. In the context of testing curve analysis, probably we need to test `CurveAnalysis._format_data` and fit curve calculation in `CurveAnalysis._run_analysis`, i.e. y values of formatted and fitted subset. Since these calculation is done behind the scene and preparation of the reference data is bit tough. For now we can go as-is and update the test when we come up with nice solution.